extensions/amp-story/1.0/amp-story-unsupported-browser-layer.js

```diff
@@ -40,66 +31,11 @@ const renderUnsuportedBrowserLayerElement = (element) => (
 );
 
 /**
- * Unsupported browser layer UI.
+ * @param {!Element} context
+ * @param {function(Event):void} continueAnyway
+ * @return {!Element}
  */
-export class UnsupportedBrowserLayer {
-  /**
-   * @param {!Window} win
-   */
-  constructor(win) {
-    /** @private @const {!Window} */
-    this.win_ = win;
-
-    /** @private {?Element} */
-    this.root_ = null;
-
-    /** @private {?Element} */
-    this.element_ = null;
-
-    /** @private {?Element} */
-    this.continueButton_ = null;
-
-    /** @private @const {!./amp-story-store-service.AmpStoryStoreService} */
-    this.storeService_ = getStoreService(this.win_);
-  }
-
-  /**
-   * Builds and appends the component in the story.
-   * @return {*} TODO(#23582): Specify return type
-   */
-  build() {
-    if (this.root_) {
-      return this.root_;
-    }
-    this.root_ = this.win_.document.createElement('div');
-    this.element_ = renderUnsuportedBrowserLayerElement(this.win_.document);
-    createShadowRootWithStyle(this.root_, this.element_, CSS);
-    this.continueButton_ = this.element_./*OK*/ querySelector(
-      `.${CONTINUE_ANYWAY_BUTTON_CLASS}`
-    );
-    this.continueButton_.addEventListener('click', () => {
-      this.storeService_.dispatch(Action.TOGGLE_SUPPORTED_BROWSER, true);
-    });
-    return this.root_;
-  }
-
-  /**
-   * Returns the unsupported browser componenet
-   * @return {?Element} The root element of the componenet
-   */
-  get() {
-    if (!this.root_) {
-      this.build();
-    }
-    return this.root_;
-  }
-
-  /**
-   * Removes the entire layer
-   */
-  removeLayer() {
-    if (this.root_) {
-      removeElement(this.root_);
-    }
-  }
+export function renderUnsupportedBrowserLayer(context, continueAnyway) {
```

Comment:
By the same token, we don't need a class that holds state. We only return the shadow root that holds the rendered layer.